src/main/java/com/mykaarma/urlshortener/service/UrlService.java

```diff
@@ -151,13 +137,13 @@ private UrlDetails createShortUrl(String shortUrlDomain, String longUrl, Date ex
 		else{
 			urlPrefix = "";
 		}
-
 		String shortUrl = shortUrlDomain + urlPrefix + shortUrlHash;
 
 		shortUrlDetails.setShortUrl(shortUrl);
 
 		try {
-			urlRepository.saveUrl(shortUrlDetails);
+			shortUrlDatabaseAdapter.saveUrl(shortUrlDetails);
+			hashArchiveAdapter.addHashToArchive(shortUrlHash);
```

Comment:
@vishal2295 I propose that we interchange these two lines. 

Rationale:
Is it possible that sometime the call to `shortUrlDatabaseAdapter.saveUrl(shortUrlDetails);` succeeds and `hashArchiveAdapter.addHashToArchive(shortUrlHash);` fails? One basic case I can think of is when we'd have archived some hashes, and the main collection won't throw unique key exception, while the HashArchive would.

The downside to saving in the archive first is that we may end up wasting the hash if we fail to actually create shortUrl from the same. (Which I think is acceptable, and we can add code to remove from pool in the catch block if we want)

Reply:
We talked. Even though the call to `addHashToArchive` is sync now, it may be async in the future. Thus, there is no way to guarantee 100% addition to HashArchive. We can consider having some kind of reconciliation job for this one, but only if this problem actually impacts us. For now, we don't want to interchange these two lines.